ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,135 @@
+package ch.epfl.ts.evaluation
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.{ActorRef, Cancellable}
+import ch.epfl.ts.component.{ComponentRegistration, Component, ComponentRef, StartSignal, StopSignal}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Evaluates the performance of traders by total returns, volatility, draw down and sharp
+  *
+  * To use this class, redirect all previous connections into trader to instances
+  * of this class.
+  *
+  * @param trader the reference to the real trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the intial seed money
+  * @param period the time period to compute performance
+  */
+class Evaluator(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: FiniteDuration) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
```

Comment:
I'm not sure if we need to use a broker in order to evaluate the performance of traders. Anyway,  using the wallet in broker makes the implementation of `Evaluator` much more complex(share data across actors!), which doesn't justify the approach you suggested.

One improvement I see is to abstract a concept Wallet, and add one method to calculate its total value in a given currency(with the given exchange rates). Then it can save about 5 lines of code in current implementation.